### PR TITLE
list videos design template

### PR DESCRIPTION
**Pull request by R-Bear**
# Design List Videos for user

## Show videos
- [x] videos.scss 
- [x] responsive

## routes 
- show: `/videos/show`
## views
- directory: `videos/show.html.erb`

### CODE REVIEW

File: pj_movie/config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,6 @@
   root 'static_pages#home'
   get 'static_pages/home'
   get 'static_pages/help'
+  get 'videos/show'
```
Comment (N310) [inline pj_movie/config/routes.rb]: you should use `index` action.

File: pj_movie/spec/controllers/videos_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe VideosController, type: :controller do
```
Comment (N310) [inline pj_movie/spec/controllers/videos_controller_spec.rb]: we should use request spec for controller test.

- Reference
https://relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec

File: pj_movie/app/assets/stylesheets/videos.scss [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        flex-direction: row;
+        flex-wrap: wrap;
+        .box_text{
+          display: flex;
+          flex-direction: column;
+          flex: 3;
+          margin: 10px;
+          box-sizing: border-box;
+          video{
+          height: 180px;
```
Comment (N310) [inline pj_movie/app/assets/stylesheets/videos.scss]: you shouldn't change aspect rate of movies.

File: pj_movie/app/assets/stylesheets/admin/movies.scss [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        width: 100%;
+        border-bottom: 3px  solid #e2dddd;
+        font-weight: bold;
+        font-size: 24px;
+        margin: 15px 0;
+        p {
+          margin: 0;
+        }
+      }
+      .list_vdeos {
```
Comment (N310) [inline pj_movie/app/assets/stylesheets/admin/movies.scss]: I think that side spaces is too large in about 641px ~ 675px.
Please you check.

File: pj_movie/app/assets/stylesheets/movies.scss [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        width: 100%;
+        border-bottom: 3px  solid #e2dddd;
+        font-weight: bold;
+        font-size: 24px;
+        margin: 15px 0;
+        p {
+          margin: 0;
+        }
+      }
+      .list_vdeos {
```
Comment (N310) [inline pj_movie/app/assets/stylesheets/movies.scss]: I think that side spaces is too large in about 641px ~ 675px.
Please you check.

File: pj_movie/config/environments/development.rb [hunk 6]
```diff
@@ -9,7 +9,7 @@
   # Do not eager load code on boot.
   config.eager_load = false
 
-  # Show full error reports.
+  # Index full error reports.
```
Comment (N310) [inline pj_movie/config/environments/development.rb]: Do you need this change ?

File: pj_movie/config/environments/test.rb [hunk 7]
```diff
@@ -18,12 +18,12 @@
...
   }
 
-  # Show full error reports and disable caching.
+  # Index full error reports and disable caching.
   config.consider_all_requests_local       = true
   config.action_controller.perform_caching = false
 
   # Raise exceptions instead of rendering exception templates.
-  config.action_dispatch.show_exceptions = false
+  config.action_dispatch.index_exceptions = false
```
Comment (N310) [inline pj_movie/config/environments/test.rb]: Do you need this change ?

File: pj_movie/config/routes.rb [hunk 8]
```diff
@@ -3,4 +3,17 @@
   root 'static_pages#home'
   get 'static_pages/home'
   get 'static_pages/help'
+  resources :movies do
+    collection do
+      get 'index'
+    end
+  end
```
Comment (N310) [inline pj_movie/config/routes.rb]: you can change this block to `resources :movies, only: [:index]`

File: pj_movie/config/routes.rb [hunk 9]
```diff
@@ -3,4 +3,17 @@
...
+  namespace :admin do
+    resources :movies do
+      collection do
+        get 'index'
+      end
+    end
```
Comment (N310) [inline pj_movie/config/routes.rb]: you can change this are to `resources :movies, only: [:index]`

File: pj_movie/app/assets/stylesheets/admin/movies.scss [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+        flex-direction: row;
+        flex-wrap: wrap;
+        justify-content: space-between;
+        margin-top: 15px;
+        .box_text {
+          margin: 10px;
+          box-sizing: border-box;
+          width: 30%;
+          margin-top: 20px;
+          video{
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/movies.scss]: you should add one more space before "{"

File: pj_movie/app/assets/stylesheets/admin/movies.scss [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+            margin-right: 5px;
+          }
+          .video_discribe span:first-child {
+            font-weight: bold;
+            width: 100%;
+            margin-top: 10px;
+          }
+        }
+      }
+      .more{
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/movies.scss]: also should add more space
.more{ => .more {

File: pj_movie/app/assets/stylesheets/admin/movies.scss [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+              font-display: row;
+              flex-wrap: wrap;
+              font-size: 16px;
+            }
+            .video_discribe span {
+              flex-wrap: wrap;
+              margin-left: 5px;
+              margin-right: 5px;
+            }
+            .video_discribe span:first-child{
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/movies.scss]: also should add more space

File: pj_movie/app/assets/stylesheets/admin/movies.scss [hunk 13]
```diff
@@ -0,0 +1,136 @@
+//Admin list videos Index video style
+.adminvideo_body {
+  display: flex;
+  flex-direction: column;
+  font-size:16px;
+  font-family: Roboto;
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/movies.scss]: I think this font-family no need, we already use in custom.scss

File: pj_movie/app/assets/stylesheets/admin/password_resets.scss [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+      flex-direction: column;
+      align-items: center;
+      justify-content: center;
+      padding: 0;
+      .forgot_password {
+        display: flex;
+        flex-direction: column;
+        justify-content: left;
+        font-size: 16px;
+        font-family: Roboto;
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/password_resets.scss]: font-family, we already have in custom.scss, so no need here

File: pj_movie/app/assets/stylesheets/admin/password_resets.scss [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+          display: flex;
+          flex-direction: column;
+          justify-content: center;
+          align-items: center;
+          text-align: left;
+          font-size: 16px;
+          border: none;
+          padding: 0;
+          width: 95%;
+          font-family: Roboto;
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/password_resets.scss]: should remove this line

File: pj_movie/app/assets/stylesheets/movies.scss [hunk 16]
```diff
@@ -0,0 +1,139 @@
+// Index video style
+.video_body {
+  display: flex;
+  flex-direction: column;
+  font-size:16px;
+  font-family: Roboto;
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/movies.scss]: should remove this line

File: pj_movie/app/assets/stylesheets/movies.scss [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+        display: flex;
+        flex-direction: row;
+        flex-wrap: wrap;
+        justify-content: space-between;
+        margin-top: 15px;
+        .box_text{
+          box-sizing: border-box;
+          width: 30%;
+          margin-top: 20px;
+          video{
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/movies.scss]: you should add more space

File: pj_movie/app/assets/stylesheets/password_resets.scss [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+      flex-direction: column;
+      align-items: center;
+      justify-content: center;
+      padding: 0;
+      .forgot_password {
+        display: flex;
+        flex-direction: column;
+        justify-content: left;
+        font-size: 16px;
+        font-family: Roboto;
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/password_resets.scss]: I think no need this font

File: pj_movie/app/views/layouts/application.html.erb [hunk 19]
```diff
@@ -1,9 +1,9 @@
 <!DOCTYPE html>
 <html>
   <head>
-  <title><%= yield(:title) %> | Project Movie</title>
-  <%= render 'layouts/rails_default' %>
-  <%= render 'layouts/shim' %>
+    <title><%= yield(:title) %> | Project Movie</title>
```
Comment (loemhuolay) [inline pj_movie/app/views/layouts/application.html.erb]: you should remove "|" before Project
Ex: %> | Project Movie</title>        =>            %> Project Movie </title>

File: pj_movie/app/views/admin/password_resets/edit.html.erb [hunk 20]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title><%= yield(:title) %> | Project Movie</title>
```
Comment (loemhuolay) [inline pj_movie/app/views/admin/password_resets/edit.html.erb]: you should change this line 
Ex:  <title><%= yield(:title) %> | Project Movie</title> to 
 <title><%= yield(:title) %> Project Movie </title>

File: pj_movie/app/views/admin/password_resets/new.html.erb [hunk 21]
```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title><%= yield(:title) %> | Project Movie</title>
```
Comment (loemhuolay) [inline pj_movie/app/views/admin/password_resets/new.html.erb]: also check this line  <title><%= yield(:title) %> | Project Movie</title>

File: pj_movie/app/assets/stylesheets/admin/movies.scss [hunk 22]
```diff
@@ -0,0 +1,144 @@
...
+          .video_discribe span:first-child {
+            font-weight: bold;
+            width: 100%;
+            margin-top: 10px;
+          }
+          .movies_tag {
+            border: 1px solid #e2dddd;
+            margin: 5px 0;
+            padding: 5px;
+            span{
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/movies.scss]: please check  span{

File: pj_movie/app/assets/stylesheets/custom.scss [hunk 23]
```diff
@@ -83,7 +83,7 @@ textarea {
           }
         }
       }
-      &:hover .dropdown_content {
+      &:hover .dropdown_content{
```
Comment (N310) [inline pj_movie/app/assets/stylesheets/custom.scss]: You shouldn't remove space.

File: pj_movie/app/assets/stylesheets/admin/top_page.scss [hunk 24]
```diff
@@ -162,14 +162,11 @@
...
-.container-admin {
-  display: flex;
-  justify-content: center;
-  align-items: center;
+.container.container-admin {
   margin-left: 246px;
   padding-top: 57px;
   padding-bottom: 40px;
-  background-color: #fafafa;
+  height: 100vh;
```
Comment (N310) [inline pj_movie/app/assets/stylesheets/admin/top_page.scss]: Do you need this line?

**Pull request merged**